extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -105,76 +236,48 @@ public Result<CreateEntryResponsePayload> setSecret(@NotNull String key, @NotNul
 
         try (var response = httpClient.execute(request)) {
             if (response.isSuccessful()) {
-                var responseBody = Objects.requireNonNull(response.body()).string();
+                if (response.body() == null) {
+                    return Result.failure("Setting secret %s returned empty body".formatted(key));
+                }
+                var responseBody = response.body().string();
                 var responsePayload =
                         objectMapper.readValue(responseBody, CreateEntryResponsePayload.class);
                 return Result.success(responsePayload);
             } else {
-                return Result.failure(String.format(CALL_UNSUCCESSFUL_ERROR_TEMPLATE, response.code()));
+                return Result.failure("Failed to set secret %s with status %d".formatted(key, response.code()));
             }
         } catch (IOException e) {
-            return Result.failure(e.getMessage());
+            var errMsg = "Failed to set secret %s with reason: %s".formatted(key, e.getMessage());
+            monitor.warning(errMsg, e);
+            return Result.failure(errMsg);
         }
     }
 
     public Result<Void> destroySecret(@NotNull String key) {
         var requestUri = getSecretUrl(key, VAULT_SECRET_METADATA_PATH);
-        var headers = getHeaders();
         var request = new Request.Builder().url(requestUri).headers(headers).delete().build();
 
         try (var response = httpClient.execute(request)) {
             return response.isSuccessful() || response.code() == HTTP_CODE_404
                     ? Result.success()
-                    : Result.failure(String.format(CALL_UNSUCCESSFUL_ERROR_TEMPLATE, response.code()));
+                    : Result.failure("Failed to destroy secret %s with status %d".formatted(key, response.code()));
```

Comment:
The secret key should not be recorded. The client knows which key it asked for and can handle it appropriately